hubbard/grid.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import sisl
+
+
+def real_space_grid(geometry, v, grid_unit, xmin, xmax, ymin, ymax, z=1.1, mode='wavefunction', **kwargs):
+    """
+    Parameters
+    ----------
+    g : sisl.Geometry
+    v : numpy.array
+    grid_unit: 3-array-like
+    mode: str, optinoal
+        to build the grid from sisl.electron.wavefunction object (``mode=wavefunction``)
+        or from the sisl.physics.DensityMatrix (``mode=`charge`), e.g. for charge-related plots
+    sc: sisl.SuperCell, optional
+        super cell, defaults to the sc saved in the geometry
+    """
+    # Create a temporary copy of the geometry
+    g = geometry.copy()
+
+    # Set new sc to create real-space grid
+    sc = sisl.SuperCell([xmax-xmin, ymax-ymin, 1000], origin=[0, 0, -z])
+    g.set_sc(sc)
+
+    # Move geometry within the supercell
+    g = g.move([-xmin, -ymin, -np.amin(g.xyz[:, 2])])
+    # Make z~0 -> z = 0
+    g.xyz[np.where(np.abs(g.xyz[:, 2]) < 1e-3), 2] = 0
```

Comment:
This is for our own purposes actually, since we typically plot the geometry backbone on top of the grid to have both the LDOS and the geometry in the plot. With this changes we have the same sizes for the boxes of the geometry and the LDOS colormap Of course the user can have their own function to build the grid.